openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -63,43 +73,72 @@ def get_next_query_params(self, prev_query_params=None, **kwargs):
-            # more info in https://github.com/WordPress/openverse-catalog/issues/684
-            return None
-        license_url = data.get("license_url")
-        license_info = get_license_info(license_url=license_url)
-        if license_info == NO_LICENSE_FOUND:
-            return None
-        record_data = {k: data[k] for k in data.keys() if k != "license_url"}
-        record_data["license_info"] = license_info
-        return record_data
+        raise NotImplementedError("TSV files from AWS S3 processed in postgres.")
 
     def get_media_type(self, record):
         # This provider only supports Images via S3, though they have some audio files
         # on site and in the API.
-        return "image"
+        # I'm using a constant here, which maybe should just be a class constant, but
+        # it's there to work with the static methods.
+        return MEDIA_TYPE
 
     def endpoint(self):
         raise NotImplementedError("Normalized TSV files from AWS S3 means no endpoint.")
 
+    @staticmethod
+    def load_transformed_data(
+        intermediate_table: str,
+        page_length: int,
+        postgres_conn_id=POSTGRES_CONN_ID,
+        sql_template_file_name="transformed_table.template.sql",
+        identifier: str = None,
+    ):
+        pg = PostgresHook(postgres_conn_id)
+        sql_template = (SCRIPT_DIR / sql_template_file_name).read_text()
+        total_records = 0
+        # get the number of "pages" in the photos table
+        max_id = pg.get_records("SELECT max(photo_id) FROM inaturalist.photos")[0][0]
+        total_pages = int(max_id / page_length) + 1
+        logger.info(f"Loading {intermediate_table}...")
+        logger.info(f"Max photo_id={max_id}. Transform {page_length} ids / iteration.")
+        # then cycle through the pages
+        for page_start in range(0, max_id, page_length):
+            page_end = page_start + page_length - 1
+            page_log = f"page {int(page_start / page_length) + 1} of {total_pages}"
+            logger.info(f"Starting at photo_id {page_start}, {page_log}")
+            (transformed_records, max_id_loaded) = pg.get_records(
+                sql_template.format(
+                    intermediate_table=intermediate_table,
+                    page_start=page_start,
+                    page_end=page_end,
+                )
+            )[0]
+            logger.info(
+                f"Inserted {transformed_records} into {intermediate_table}. "
+                f"Last photo_id loaded was {max_id_loaded}, as of {page_log}."
+            )
+            # Run standard cleaning on this batch of records
+            sql.clean_transformed_provider_s3_data(
+                identifier=identifier,
+                postgres_conn_id=postgres_conn_id,
+            )
+            # Add transformed records to the target catalog image table.
+            total_records += sql.upsert_records_to_db_table(
```

Comment:
Can these be split into individual tasks? I _think_ we could even still use dynamic task mapping with a resulting [TaskGroup](https://docs.astronomer.io/learn/task-groups#generate-task-groups-dynamically-at-runtime)

Reply:
Ooooh! I didn't know about loops within DAGs / across tasks. Neat! I'll take a look.

Reply:
So, I guess I did know about dynamic tasks because we're using them for [pulling raw inaturalist tables](https://github.com/WordPress/openverse-catalog/blob/382192584f6004537985f754f991e4d51cdbfa85/openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py#L146), but I don't think they would work here for pages/chunks of iNaturalist photos, in a single dag. (More on this in a comment below.)

Reply:
Just linking my [other comment](https://github.com/WordPress/openverse-catalog/pull/745/files/5437d7ba599d28c3be8afbcf5a2f5a5b07847106#r1054917849) for clarity that this can definitely be something we look at in a future iteration :)